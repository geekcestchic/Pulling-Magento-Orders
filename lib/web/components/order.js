import React from 'react'
import style from '../styles/style'

function ExtraData (props) {
  const order = props.order
  const weight = parseInt(order.weight, 10)
  return (
    <div>
      <div className="modale-background" style={style.modal}/>
      <div className="detail-modal" style={style.detail}>
        <div className="close" style={style.close}>X</div>
        <div>Updated At: {order.updated_at}</div>
        <div>Customer Email: {order.customer_email}</div>
        <div>Order ID: {order.order_id}</div>
        <div>Postcode: {order.postcode}</div>
        <div>Weight: {weight}</div>
      </div>
    </div>
  )
}

class Order extends React.Component {
  constructor(props) {
    super(props)
    this.state = { expanded: false }
  }
  expand () {
    if (this.state.expanded === false) {
      this.setState({
        expanded: true
      })
    } else if (this.state.expanded === true) {
      this.setState({
        expanded: false
      })
    }

  }
  render () {
    const {order} = this.props
    const fullName = (order.firstname || '') + ' ' + (order.lastname || '')
    const total = order.total_paid ? 'Paid ' + parseInt(order.total_paid, 10) + ' ' + order.order_currency_code : 'Not paid'
    return (
      <div style={style.order} className="order" onClick={this.expand.bind(this)}>
        <div>Order For - {fullName}</div>
        <div>Payment Status: {total}</div>
        {this.state.expanded ? <ExtraData order={this.props.order}/> : <button>Click to Expand</button> }
      </div>
    )
  }
}

export default Order
