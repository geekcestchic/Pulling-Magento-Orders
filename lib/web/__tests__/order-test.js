import React from 'react'
import Order from '../components/order'
import renderer from 'react-test-renderer'
import order from './order-object'

it('responds to click events correctly', () => {
  const component = renderer.create(
    <Order order={order}></Order>
  )
  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()

  // manually trigger the callback
  tree.props.onClick()
  // re-rendering
  tree = component.toJSON()
  expect(tree).toMatchSnapshot()

  // // manually trigger the callback
  // tree.props.onClick()
  // // re-rendering
  // tree = component.toJSON()
  // expect(tree).toMatchSnapshot()
})
