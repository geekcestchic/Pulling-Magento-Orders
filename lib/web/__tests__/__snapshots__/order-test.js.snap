exports[`test responds to click events correctly 1`] = `
<div
  className="order"
  onClick={[Function bound expand]}
  style={
    Object {
      "border": "1px solid black",
      "borderRadius": 10,
      "cursor": "pointer",
      "display": "inline-block",
      "fontFamily": "arial",
      "height": 50,
      "margin": "2%",
      "padding": 20,
      "textAlign": "center",
      "width": "40%"
    }
  }>
  <div>
    Order For - 
    Adham Bard
  </div>
  <div>
    Payment Status: 
    Paid 3748 EUR
  </div>
  <button>
    Click to Expand
  </button>
</div>
`;

exports[`test responds to click events correctly 2`] = `
<div
  className="order"
  onClick={[Function bound expand]}
  style={
    Object {
      "border": "1px solid black",
      "borderRadius": 10,
      "cursor": "pointer",
      "display": "inline-block",
      "fontFamily": "arial",
      "height": 50,
      "margin": "2%",
      "padding": 20,
      "textAlign": "center",
      "width": "40%"
    }
  }>
  <div>
    Order For - 
    Adham Bard
  </div>
  <div>
    Payment Status: 
    Paid 3748 EUR
  </div>
  <div>
    <div
      className="modale-background"
      style={
        Object {
          "backgroundColor": "grey",
          "fontFamily": "arial",
          "height": "100%",
          "left": 0,
          "opacity": 0.5,
          "position": "fixed",
          "top": 0,
          "width": "100%"
        }
      } />
    <div
      className="detail-modal"
      style={
        Object {
          "background": "white",
          "border": "2px solid black",
          "height": 200,
          "left": "50%",
          "margin": "-100px 0 0 -200px",
          "padding": 30,
          "position": "fixed",
          "textAlign": "left",
          "top": "50%",
          "width": 400,
          "zIndex": 10
        }
      }>
      <div
        className="close"
        style={
          Object {
            "height": 40,
            "paddingTop": 10,
            "position": "absolute",
            "right": 0,
            "textAlign": "center",
            "top": 0,
            "width": 40
          }
        }>
        X
      </div>
      <div>
        Updated At: 
        2015-10-26 15:35:09
      </div>
      <div>
        Customer Email: 
        1444656004@example.com
      </div>
      <div>
        Order ID: 
        1
      </div>
      <div>
        Postcode: 
        80801
      </div>
      <div>
        Weight: 
        3
      </div>
    </div>
  </div>
</div>
`;
